packages/itwinui-css/src/side-navigation/side-navigation.scss

```diff
@@ -161,8 +161,36 @@ $iui-side-navigation-icon-margins: calc(1.5 * var(--iui-size-m));
   }
 }
 
+//animating technique from https://css-tricks.com/css-grid-can-do-auto-height-transitions/
+.iui-side-navigation-submenu-wrapper {
+  display: grid;
+  grid-template-columns: 0fr;
+  overflow: hidden;
+  z-index: -1;
+  justify-content: end;
+  justify-items: end;
```

Comment:
Answering your earlier question in a thread:

> > to keep the child from moving, you'll need to remove `transform: translateX(-100%)` from the parent and add `justify-content: end` + `justify-items: end`.
>
> Wow thanks! That works! But I am so confused. How come in expandable block which I feel was exact logic, worked with `transform: translateX(-100%)` but not here and we need the `justify-content` + `items` here?

This is very similar to `align-self: end` added in #1396.

<img width="279" alt="" src="https://github.com/iTwin/iTwinUI/assets/9084735/5b0ebdaf-a151-430f-9713-1a1f45254c8b">

Maybe `align-content: end` would have also helped on the parent there. Not sure. And the `transform: translate` part was just to work around the padding issue from https://github.com/iTwin/iTwinUI/pull/1396#discussion_r1258323140.

There is also a difference of horizontal vs vertical, so the code is expected to be different.

I'm also realizing that sidenav has `overflow: hidden`, so this whole grid technique might even be unnecessary and could be replaced with a simple `transform: translateX(-100%)`. See https://web.dev/building-a-sidenav-component/#transforms.